src/api/protectedApiClient.ts

```diff
@@ -39,12 +40,20 @@ export interface ProtectedApiClient {
   readonly setContactInfo: (request: ContactInfo) => Promise<void>;
   readonly changeAccountEmail: (request: ChangeEmailRequest) => Promise<void>;
   readonly getEventRegistrations: (eventId: number) => Promise<Registration[]>;
+  readonly eventCreate: (
+    request: NewEventInformation,
+  ) => Promise<EventInformation>;
+  readonly eventEdit: (
+    id: number,
+    request: NewEventInformation,
```

Comment:
⛏️ Maybe we should have an EditEventInformation which just points to the same type, i.e.
`type EditEventInformation = NewEventInformation`

Reply:
It seems a little weird to have two types that are really just one type. Can we rename NewEventInformation to RequestEventInformation instead?

Reply:
elsewhere I've used a `interface XResponse extends XRequest` pattern for adding an ID or other extra fields only in the response, not sure if that's entirely applicable here